internal/pkg/model/relation.go

```diff
@@ -41,13 +42,12 @@ func (t RelationType) Type() RelationType {
 // Relation between objects, eg. config <-> config.
 type Relation interface {
 	Type() RelationType
-	Desc() string                             // human-readable description
-	Key() string                              // unique key for sorting and comparing
-	ParentKey(relationOwner Key) (Key, error) // if relation type is parent <-> child, then parent key is returned, otherwise nil
-	OtherSideKey(owner Key) Key               // get key of the other side
-	IsDefinedInManifest() bool                // if true, relation will be present in the manifest
-	IsDefinedInApi() bool                     // if true, relation will be present in API calls
-	NewOtherSideRelation(owner Key) Relation  // create the new other side relation, for example VariablesFor -> VariablesFrom
+	Desc() string                                 // human-readable description
+	Key() string                                  // unique key within the object on which the relation is defined, for sorting and comparing
+	ParentKey(relationDefinedOn Key) (Key, error) // if relation type is parent <-> child, then parent key is returned, otherwise nil
+	IsDefinedInManifest() bool                    // if true, relation will be present in the manifest
+	IsDefinedInApi() bool                         // if true, relation will be present in API calls
+	NewOtherSideRelation(relationDefinedOn Object, allObjects *StateObjects) (otherSide Key, relation Relation, err error)
```

Comment:
Namiesto `OtherSideKey`, a `NewOtherSideRelation`, je uz iba metoda `NewOtherSideRelation`:
- obe hodnoty spolu suvisia, `key` hovori, ktory objekt je na druhej strane, a `relation` aka je relation na druhej strane, definovana na tomto objekte
- metoda dostava naviac `allObjects`, ak relacia potrebuje aj info o inych objektoch, ... napr. aky je `parent`, ... vyuziva sa to pri` variablesValuesFor`, ... aby nemuselo byt ulozene duplicitne ID v `manifest.json` (vid ostatne komentare).